bloomfilter/Data/BloomFilter/Mutable.hs

```diff
@@ -106,7 +97,7 @@ elem elt mb = loop (hashes mb elt)
   where mu = bitArray mb
         loop (idx':wbs) = do
           -- the index calculation works as long as sizeof(Int) >= sizeof(Hash).
-          let !idx = fromIntegral idx' `rem` size mb :: Int
+          let !idx = fromIntegral (idx' `rem` fromIntegral (size mb)) :: Int
```

Comment:
Doing these type conversions, and doing the remainder operation at the right type is important and a little subtle, and is replicated 3 times. Perhaps we should pull it out into an inlined function to do it once and be clear about the types.

e.g.
```
{-# INLINE probeIndex #-}
probeIndex :: Int -> Hash -> Int
probeIndex size h =
    fromIntegral (h `rem` fromIntegral size :: Word64) :: Int
    -- we do the remainder at type Word64, but the resulting probe index
    -- necessarily fits in an Int (even on 32bit platforms)
```
and then we could use the same function between the inserts and the queries for both mutable and immutable.